src/g_shared/a_weapons.cpp

```diff
@@ -222,6 +222,10 @@ bool AWeapon::HandlePickup (AInventory *item)
 		{
 			item->ItemFlags |= IF_PICKUPGOOD;
 		}
+		if (MaxAmount > 1) //[SP] If amount<maxamount do another pickup test of the weapon itself!
```

Comment:
This "if" statement MAY be considered vestigial. I doubt it's really needed. I left it in for sanity's sake and for speed purposes. If it is removed, then the "return true" below it should be removed as well.